mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionFactory.kt

```diff
@@ -318,12 +318,12 @@ class ConnectionFactory internal constructor(
     private class ClientKeyManager(context: Context, private val alias: String?) : X509KeyManager {
         private val context: Context = context.applicationContext
 
-        override fun chooseClientAlias(keyType: Array<String>, issuers: Array<Principal>, socket: Socket): String? {
+        override fun chooseClientAlias(keyType: Array<String>, issuers: Array<Principal>?, socket: Socket?): String? {
```

Comment:
How can you make sure `keyType` isn't null? I don't see any `@NonNull` in `X509KeyManager`.

Reply:
I reviewed the conscrypt sources:

- https://android.googlesource.com/platform/external/conscrypt/+/refs/heads/master/common/src/main/java/org/conscrypt/ConscryptEngine.java#1715
- called by https://android.googlesource.com/platform/external/conscrypt/+/refs/heads/master/common/src/main/java/org/conscrypt/NativeSsl.java#227
- variable comes from https://android.googlesource.com/platform/external/conscrypt/+/refs/heads/master/common/src/main/java/org/conscrypt/NativeSsl.java#214

I'll change it to follow the inferred types though (I just noticed when creating a new X509KeyManager subclass and doing 'Implement methods' in AS, the types are nullable).